src/lib/components/Swap/SwapButton.tsx

```diff
@@ -129,18 +132,36 @@ export default function SwapButton({ disabled }: SwapButtonProps) {
 
   //@TODO(ianlapham): add a loading state, process errors
   const setDisplayTxHash = useUpdateAtom(displayTxHashAtom)
+
   const onConfirm = useCallback(() => {
     swapCallback?.()
-      .then((transactionResponse) => {
-        // TODO(ianlapham): Add the swap tx to transactionsAtom
-        console.log(transactionResponse)
-        setDisplayTxHash(transactionResponse.hash)
+      .then((response) => {
+        setDisplayTxHash(response.hash)
+        invariant(inputCurrencyAmount && outputCurrencyAmount)
+        if (independentField === Field.INPUT) {
+          addTransaction({
+            response,
+            type: TransactionType.SWAP,
+            tradeType: TradeType.EXACT_INPUT,
+            inputCurrencyAmount,
+            outputCurrencyAmount,
+          })
+        }
+        if (independentField === Field.OUTPUT) {
+          addTransaction({
+            response,
+            type: TransactionType.SWAP,
+            tradeType: TradeType.EXACT_OUTPUT,
+            inputCurrencyAmount,
+            outputCurrencyAmount,
+          })
```

Comment:
can you just do a conditional on the single field that's different here to cut down on repetition?

Reply:
i thought of this - i think this pattern works better if we expect more fields to be added in future, aka a ternary may not resolve correctly if there are 3 fields

but, i dont see us addign a third field to swap unless some weir changes to protoocl or router are added, so ill go with your suggestion